tests/RobotFramework/tests/cumulocity/firmware/firmware_operation_child_device.robot

```diff
@@ -0,0 +1,168 @@
+
+    Sleep    3s
+    Execute Command    sudo tedge connect c8y
+    Restart Firmware plugin    #Stop and Start c8y-firmware-plugin
+    Cumulocity.Log Device Info
+
+Prerequisite Child
+    Child device delete firmware file    #Delete previous downloaded firmware file
+    Create child device    #Let mapper to create a child device
+    Validate child Name    #This is to check the existence of the bug: https://github.com/thin-edge/thin-edge.io/issues/1569
+
+Child device firmware update
+    Upload binary to Cumulocity
+    Create c8y_Firmware operation
+    Validate firmware update request
+    Child device response on update request    #Child device is sending 'executing' and 'successful' MQTT responses
+    Validate Cumulocity operation status and MO
+
+Child device firmware update with cache
+    Get timestamp of cache
+    Create c8y_Firmware operation
+    Validate firmware update request
+    Child device response on update request    #Child device is sending 'executing' and 'successful' MQTT responses
+    Validate Cumulocity operation status and MO
+    Validate if file is not newly downloaded
+
+*** Keywords ***
+Delete child related content
+    Execute Command    sudo rm -rf /etc/tedge/operations/c8y/TST*         #if folder exists, child device will be created
+    Execute Command    sudo rm -rf /etc/tedge/c8y/TST*                    #if folder exists, child device will be created
+    Execute Command    sudo rm -rf /var/tedge/file-transfer/*
+    Execute Command    sudo rm -rf /var/tedge/cache/*
+    Execute Command    sudo rm -rf /var/tedge/firmware/*
+
+Check for child related content
+    Set Device Context    ${PARENT_SN}
+    Directory Should Not Have Sub Directories    /etc/tedge/operations/c8y
+    Directory Should Not Have Sub Directories    /etc/tedge/c8y
+    Directory Should Not Have Sub Directories    /var/tedge/file-transfer
+    Directory Should Be Empty    /var/tedge/cache
+    Directory Should Be Empty    /var/tedge/firmware
+
+Set external MQTT bind address
+    Set Device Context    ${PARENT_SN}
+    Execute Command    sudo tedge config set mqtt.external.bind_address ${PARENT_IP}
+
+Set external MQTT port
+    Set Device Context    ${PARENT_SN}
+    Execute Command    sudo tedge config set mqtt.external.port 1883
+
+Restart Firmware plugin
+    ThinEdgeIO.Restart Service    c8y-firmware-plugin.service
+
+Child device delete firmware file
+    Set Device Context    ${CHILD_SN}
+    Execute Command    sudo rm -f firmware1
+
+Create child device
+    [Documentation]    FIXME: Due to the initialization bug, child device not being registered, we need to restart mapper. See the bug report XXX
+    Set Device Context    ${PARENT_SN}
+    Execute Command    mkdir -p /etc/tedge/operations/c8y/${CHILD_SN}
+    ThinEdgeIO.Transfer To Device    ${CURDIR}/c8y_Firmware    /etc/tedge/operations/c8y/${CHILD_SN}/
+    ThinEdgeIO.Restart Service    tedge-mapper-c8y
+    Cumulocity.Device Should Exist    ${CHILD_SN}
```

Comment:
I belive this sequence is causing the child device to be registered implicity resulting in the child device name having the `MQTT Device ` prefix in the name.

For example the first test run had a failure when validating the child device name:

```
TST_rev_dense_manuscript != MQTT Device TST_rev_dense_manuscript
```

Reply:
Ah horrible, race condition. That means some SmartREST message was published to the child device before `tedge-mapper-c8y` sends 101 to create a child device with appropriate inventory fragments :/

Reply:
Yep this old chestnut.

Reply:
Yeah, first `114` (declaring supported operation) is arrived, then `101` (creating a child device) is arrived in c8y. Should be opposite order.

After a couple of trials, I figure out that we don't need a restart. Instead, `sleep 3s` after `Set Device Context` helps. From this, I started feeling that the issue is in testing setup environment side. Therefore, I decided not to create an issue yet.

Reply:
> Though today a "sleep 3s" helps, tomorrow it doesn't...makes for a flakey test. I would much prefer to create a dynamic wait instead.

Definitely dynamic wait makes much more sense! Sleep with static seconds is just a workaround...